datahub-web-react/src/app/homeV3/modules/hierarchyViewModule/components/domains/hooks/useInitialDomains.ts

```diff
@@ -0,0 +1,34 @@
+import { useMemo } from 'react';
+
+import { isDomain } from '@app/entityV2/domain/utils';
+
+import { useGetSearchResultsForMultipleQuery } from '@graphql/search.generated';
+import { EntityType, FilterOperator } from '@types';
+
+export default function useInitialDomains(domainUrns: string[]) {
+    const { data, loading } = useGetSearchResultsForMultipleQuery({
+        variables: {
+            input: {
+                query: '*',
+                types: [EntityType.Domain],
+                filters: [{ field: 'urn', condition: FilterOperator.Equal, values: domainUrns }],
+                count: domainUrns.length,
+            },
+        },
+        skip: !domainUrns.length,
+        fetchPolicy: 'cache-and-network',
+    });
```

Comment:
we also have a `getEntities` endpoint that will be faster here since we have a list of urns. that way we don't have to do a trip to elasticsearch like we do here in the search, since we already have the urns we care about.

not blocking but would be a slight improvement